src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -569,7 +569,12 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         assert Thread.holdsLock(this);
         ensureOpen();
         settings = getSettings(nodeId, seed, settings);
-        String name = buildNodeName(nodeId);
+        String name;
+        if (settings.get("node.name") != null) {
```

Comment:
It does make it easy to read the test and the logs if we have full control over naming. It is not often needed I guess but for the tests above I found it incredible helpful. What is the harm when we allow to provide custom names?